### ISSUE TITLE
pq: no data source matches prefix - on UPDATE with DELETE subquery with RETURNING

### ISSUE THREAD

**Issue by insaner** [cockroachdb/cockroach#49576]
I'm trying stuff out in cockroachdb, learning the ropes still, and I'm trying to use the `RETURNING` feature of `DELETE` to then modify a row on another table, as follows:

```
UPDATE u SET c = true WHERE id = (DELETE FROM uu WHERE code = 'some code' RETURNING id);
```

But I am getting an error:
```
invalid syntax: statement ignored: at or near "from": syntax error
DETAIL: source SQL:
UPDATE u SET c = true WHERE id = (DELETE FROM uu WHERE code = 'some code'  RETURNING id)
                                         ^
HINT: try \h UPDATE
```
If I run these separately they work fine (supplying the data as needed).

I also tried:

```
WITH cu AS (DELETE FROM uu WHERE code = 'some code' RETURNING id) UPDATE u SET c = true WHERE id = cu.id
```

but I get:
```
pq: no data source matches prefix: cu
```

I know there are workarounds, but these introduce (worse) race conditions that I would like to avoid if possible. Thanks.

build:               CCL v19.2.6 @ 2020/04/06 18:05:31 (go1.12.12)

This might be related to: https://github.com/cockroachdb/cockroach/issues/43963 and https://github.com/cockroachdb/cockroach/issues/3288

**Comment by andy-kimball** [comment]
This works in CRDB and PG, it's just you don't have syntax quite right:
```
WITH cu AS (DELETE FROM uu WHERE code = 'some code' RETURNING id)
UPDATE u SET c = true FROM cu WHERE id = cu.id
```

**Comment by andy-kimball** [comment]
Closing, since this works. Please re-open if for some reason you can't get it to work.

**Comment by insaner** [comment]
Confirmed! Awesome! It works now, thanks!

It would be a great addition to the help files by the way, so someone else doesn't fall into the same trap. (Specifically how the `WITH` keyword works, I had to use postgresql documentation to even realize this was an option). Really enjoying cockroachdb, btw! Good job to all the devs and everyone involved!

**Comment by rytaft** [comment]
We have documentation [here](https://www.cockroachlabs.com/docs/v20.1/common-table-expressions.html), but it seems like we might have a documentation search bug that makes it difficult to find. That page shows up in the search results if you search for "CTE" or "Common Table Expressions", but not if you search for "WITH".

Is this the documentation you were looking for?

**Comment by insaner** [comment]
I think that's exactly it, yeah, thank you! I had looked for it (the `WITH` keyword) under the other keywords, in the left hand list under `SQL Statements`. I have to say, the documentation is a little hard to navigate, and I find myself just going to general PostgreSQL help files instead. The text in the main content is too large (and the font sizes vary too much), and a lot of the SQL examples are cut off because they don't fit in the page, for example (on a maximized window at 1920x1080 resolution). The text size for the menus is fine. There's also too much javascript fanciness that gets a little distracting. Another quick example is when I was reading the docs for `start-single-node` (as I was trying to emulate the volatility of `demo`, and have a reliable port number to connect to), I found the `type=mem` switch, but when I ran it, I got an error saying I needed to also specify the `size` attribute, which that part of the docs had not mentioned. And when I saw the section for `size` it said the default was 100%.  I program mostly in perl, and find the perldocs to be a great pattern for how to do documentation: easy to use, easy to read, and easy to quickly find solutions.

Argh, sorry, I know this is the wrong place to discuss all this stuff. In any case, I know the cockroachdb project is really young, and I'm really excited to use it, and you guys are extremely helpful (and nice!), and I am overall incredibly impressed with everything, so I hope those small pointers help out at least little to contribute to your success and wider adoption.
